lib/travis/build/git/clone.rb

```diff
@@ -17,7 +17,17 @@ def apply
 
           def clone_or_fetch
             sh.if "! -d #{dir}/.git" do
-              sh.cmd "git clone #{clone_args} #{data.source_url} #{dir}", assert: true, retry: true
+              if sparseCheckout?
+                sh.cmd "git init #{dir}", assert: true, retry: true
+                sh.cmd "git -C #{dir} git config core.sparseCheckout true", assert: true, retry: true
+                sh.cmd "echo #{sparseCheckout} >> #{dir}/.git/info/sparseCheckout", assert: true, retry: true
```

Comment:
Oh. This is `echo`. I see you are `cat`ing later.

What should the content of this file be like? I looked at https://git-scm.com/docs/git-read-tree#_sparse_checkout but it is not entirely clear.